### PR TITLE
Registration 

### LINKED ISSUE #44: Registration
## Enhance the current registration 

Be creative and enhance the suggested workflow to make it better 
___________________________________________________________________________________
### If the random surfer is a Worker
**Form 1: Personal Information** 

HEADER Highlight the Two Step Registration process  
> 1. Thumbnail for PERSONAL INFORMATION (ACTIVE)
> 2. Thumbnail for SKILLS (FROZEN)

Add the Form Content
> 1. Country of Residence 
> 2. First Name
> 3. Last Name
> 4. Email
> 5. Password
> 6. Confirm Password
> 7. Phone Number
> 8. Click Next

**The worker clicks NEXT**

**Form 2: Skill Information**
HEADER Highlight Two Step Registration process  
> 1. Thumbnail for PERSONAL INFORMATION (DONE)
> 2. Thumbnail for SKILLS (ACTIVE)

Add the Form Content
> 1. Address 
> 2. City 
> 3. Zip Code
> 4. Add SSN or Tax Identifier Number (make it optional for now)
> 5. Select Skills (from dropdown)
> 6. Select three location from where the HITs will be performed 
> 7. Add social networking profiles (optional) Linkedin/oDesk/MTurk profile
> 8. Sign UP Button

___________________________________________________________________________________

### If the random surfer is a Requestor
**Form 1: Personal Information** 

HEADER Highlight the Two Step Registration process  
> 1. Thumbnail for PERSONAL INFORMATION (ACTIVE)
> 2. Thumbnail for SKILLS (FROZEN)

Add the Form Content
> 1. Country of Residence 
> 2. First Name
> 3. Last Name
> 4. Email
> 5. Password
> 6. Confirm Password
> 7. Phone Number
> 8. Click Next

**Form 2: Skill Information**
HEADER Highlight Two Step Registration process  
> 1. Thumbnail for PERSONAL INFORMATION (Check Box DONE)
> 2. Thumbnail for SKILLS (ACTIVE)

Add the Form Content
> 1. Address 
> 2. City 
> 3. Zip Code
> 4. Add SSN or Tax Identifier Number (make it optional for now)
> 5. Select Skills (from dropdown)
> 6. Select three location from where the HITs will be performed 
> 7. Add social networking profiles (optional) Linkedin/oDesk/MTurk profile
> 8. Sign UP Button

___________________________________________________________________________________
### Folders to look at
view: `templates/authentication` 
js: `authentication (LoginController and RegisterController)` 
routes: `crowdsource.routes.js`
___________________________________________________________________________________
### Assigned 

GROUP 1
@aditimithal @rushj22 @namchester   

GROUP 2
@phiju123 

___________________________________________________________________________________
### DRI
@neilthemathguy  @dmorina @nistala

### PR DESCRIPTION
**Pull request by aditimithal**
fixes #44
Shifted from a branch to fork, updated version and errors fixed.

### CODE REVIEW

File: staticfiles/js/authentication/services/authentication.service.js [hunk 1]
```diff
@@ -59,6 +59,22 @@
...
+    * @name registerstep2
+    * @desc Try to add details of the registered user
+       * @memberOf crowdsource.authentication.services.Authentication
+    */
+    function registerstep2(address,city,zip,skillset,ssn,locations,social) {
+      return $http({
+        url: '/api/user/',
+        method: 'POST',
+        data: {
+
```
Comment (nistala) [inline staticfiles/js/authentication/services/authentication.service.js]: were not sending any data?

**Comment by nistala** [comment]
do we have this endpoint on the backend? is this currently working?

**Comment by dmorina** [comment]
The endpoint is there, used by the current register but doesnt work with this data

**Comment by nistala** [comment]
Closing PR, the backend doesn't support taking in this data yet and the frontend is not passing it back. Please reopen once fully implemented.

**Pull request closed without merging**